jenkins/opensearch/benchmark-pull-request.jenkinsfile

```diff
@@ -0,0 +1,311 @@
+            trim: true
+        )
+    }
+    triggers {
+            GenericTrigger(
+                genericVariables: [
+                    [key: 'pull_request_number', value: '$.pull_request_number'],
+                    [key: 'repository', value: '$.repository'],
+                    [key: 'DISTRIBUTION_URL', value: '$.DISTRIBUTION_URL'],
+                    [key: 'DISTRIBUTION_VERSION', value: '$.DISTRIBUTION_VERSION'],
+                    [key: 'TEST_WORKLOAD', value: '$.TEST_WORKLOAD'],
+                    [key: 'SECURITY_ENABLED', value: '$.SECURITY_ENABLED'],
+                    [key: 'SINGLE_NODE_CLUSTER', value: '$.SINGLE_NODE_CLUSTER'],
+                    [key: 'MIN_DISTRIBUTION', value: '$.MIN_DISTRIBUTION'],
+                    [key: 'MANAGER_NODE_COUNT', value: '$.MANAGER_NODE_COUNT'],
+                    [key: 'DATA_NODE_COUNT', value: '$.DATA_NODE_COUNT'],
+                    [key: 'CLIENT_NODE_COUNT', value: '$.CLIENT_NODE_COUNT'],
+                    [key: 'INGEST_NODE_COUNT', value: '$.INGEST_NODE_COUNT'],
+                    [key: 'ML_NODE_COUNT', value: '$.ML_NODE_COUNT'],
+                    [key: 'DATA_INSTANCE_TYPE', value: '$.DATA_INSTANCE_TYPE'],
+                    [key: 'DATA_NODE_STORAGE', value: '$.DATA_NODE_STORAGE'],
+                    [key: 'ML_NODE_STORAGE', value: '$.ML_NODE_STORAGE'],
+                    [key: 'JVM_SYS_PROPS', value: '$.JVM_SYS_PROPS'],
+                    [key: 'ADDITIONAL_CONFIG', value: '$.ADDITIONAL_CONFIG'],
+                    [key: 'USE_50_PERCENT_HEAP', value: '$.USE_50_PERCENT_HEAP'],
+                    [key: 'USER_TAGS', value: '$.USER_TAGS'],
+                    [key: 'WORKLOAD_PARAMS', value: '$.WORKLOAD_PARAMS'],
+                    [key: 'TEST_PROCEDURE', value: '$.TEST_PROCEDURE'],
+                    [key: 'EXCLUDE_TASKS', value: '$.EXCLUDE_TASKS'],
+                    [key: 'INCLUDE_TASKS', value: '$.INCLUDE_TASKS'],
+                    [key: 'CAPTURE_NODE_STAT', value: '$.CAPTURE_NODE_STAT'],
+                    [key: 'TELEMETRY_PARAMS', value: '$.TELEMETRY_PARAMS']
+                ],
+                tokenCredentialId: 'jenkins-pr-benchmark-generic-webhook-token',
+                causeString: 'Triggered by comment on PR on OpenSearch core repository',
+                printContributedVariables: true,
+                printPostContent: true
+            )
+    }
+
+    stages {
+        stage('validate-and-set-parameters') {
+            agent { label AGENT_LABEL }
+            steps {
+                script {
+                    if (DISTRIBUTION_URL == '' || DISTRIBUTION_VERSION == '') {
+                        currentBuild.result = 'ABORTED'
+                        error("Benchmark Tests failed to start. Provide DISTRIBUTION_URL and DISTRIBUTION_VERSION to run tests")
+                    }
+                    env.ARCHITECTURE = "x64"
```

Comment:
While using manifest this is extracted from the build manifest file but while providing the tarball url the architecture needs to be provided, for now I am restricting it to x64 tarballs since they are most commonly used.

Reply:
Got it.